span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java

```diff
@@ -127,23 +146,80 @@ private Callable<RawSpan> getRawSpanNormalizerCallable(
           .normalize(jaegerSpan, tenantIdHandler.getTenantIdProvider().getTenantIdTagKey())
           .ifPresent(rawSpanBuilder::setResource);
 
-      // redact PII tags, tag comparisons are case insensitive
+      // redact PII tags, tag comparisons are case insensitive (Resource tags are skipped)
+      sanitiseSpan(rawSpanBuilder);
+
+      // build raw span
+      RawSpan rawSpan = rawSpanBuilder.build();
+      if (LOG.isDebugEnabled()) {
+        logSpanConversion(jaegerSpan, rawSpan);
+      }
+
+      return rawSpan;
+    };
+  }
+
+  private void sanitiseSpan(Builder rawSpanBuilder) {
+
+    try {
       var attributeMap = rawSpanBuilder.getEvent().getAttributes().getAttributeMap();
       Set<String> tagKeys = attributeMap.keySet();
 
       AtomicReference<Boolean> containsPIIFields = new AtomicReference<>();
       containsPIIFields.set(false);
 
-      tagKeys.stream()
-          .filter(tagKey -> tagsToRedact.contains(tagKey.toUpperCase()))
-          .peek(tagKey -> containsPIIFields.set(true))
-          .forEach(
-              tagKey ->
+      try {
+        tagKeys.stream()
+            .filter(tagKey -> tagKeysToRedact.contains(tagKey.toUpperCase()))
+            .peek(tagKey -> containsPIIFields.set(true))
+            .forEach(
+                tagKey -> {
+                  spanAttributesRedactedCounters
```

Comment:
Do we really need this metric?

Reply:
It's for visibility just like the number of spans being dropped. We are not doing anything for that too.

Reply:
Are we going to set any alerts based on this metric? What I want to know is how will this metric be useful to us? What's an acceptable value of this metric? As I see it, having a flag that says "spansRedacted=true/false" in the past x hours is as useful as this metric, isn't it?

Reply:
>It's for visibility just like the number of spans being dropped.

The diff. is, I can set a threshold and then set alerts based on that. Because for example if my application is dropping say 500 spans/m it can be OKAY for me. However, if my app is redacting even 1 tag in the entire day it's unacceptable. Just my thoughts, please discuss with the team as well.

Reply:
hmmm okay.